WordPress/Classes/ViewRelated/Menus/MenusViewController.m

```diff
@@ -0,0 +1,945 @@
+            newMenu.name = [self generateIncrementalMenuName];
+        } else {
+            newMenu.name = menuToSave.name;
+        }
+        
+        Menu *defaultMenu = menuToSave;
+        // We'll save the newMenu instead.
+        menuToSave = newMenu;
+        // Use the items the user customized on the Default Menu as the items on the newMenu to save.
+        menuToSave.items = defaultMenu.items;
+        
+        // Reset the Default Menu.
+        defaultMenu.items = nil;
+        defaultMenu.name = [Menu defaultMenuName];
+        
+        // Add and select the new Menu in the UI.
+        self.selectedMenuLocation.menu = menuToSave;
+        [self.headerViewController addMenu:menuToSave];
+        [self.headerViewController setSelectedMenu:menuToSave];
+        [self setViewsWithMenu:menuToSave];
+    }
+    
+    __weak __typeof(self) weakSelf = self;
+    
+    void(^toggleIsSaving)(BOOL) = ^(BOOL saving) {
+        // Disable user interaction while we are processing the save.
+        weakSelf.scrollView.userInteractionEnabled = !saving;
+        // Toggle the detailsView button for "Saving...".
+        weakSelf.isSaving = saving;
+    };
+    
+    void(^failureToSave)(NSError *) = ^(NSError *error) {
+        toggleIsSaving(NO);
+        // Present the error message.
+        NSString *errorTitle = NSLocalizedString(@"Error Saving Menu", @"Menus error title for a menu that received an error while trying to save a menu.");
+        [WPError showNetworkingAlertWithError:error title:errorTitle];
+    };
+    
+    void(^updateMenu)() = ^() {
+        [weakSelf.menusService updateMenu:menuToSave
+                                  forBlog:weakSelf.blog
+                                  success:^() {
+                                      // Refresh the items stack since the items may have changed.
+                                      [WPAppAnalytics track:WPAnalyticsStatMenusSavedMenu withBlog:self.blog];
+                                      [weakSelf.itemsViewController reloadItems];
+                                      toggleIsSaving(NO);
+                                      [weakSelf setNeedsSave:NO forMenu:nil significantChanges:NO];
+                                  }
+                                  failure:failureToSave];
+    };
+    
+    toggleIsSaving(YES);
+    
+    if (menuToSave.menuID.integerValue == 0) {
+        // Need to create the menu first.
+        [self.menusService createMenuWithName:menuToSave.name
+                                         blog:self.blog
+                                      success:^(NSNumber *menuID) {
+                                          // Set the new menuID and continue the update.
+                                          [WPAppAnalytics track:WPAnalyticsStatMenusCreatedMenu withBlog:self.blog];
+                                          menuToSave.menuID = menuID;
+                                          updateMenu();
```

Comment:
Seems a bit strange that the service wouldn't just handle updating the menu in question with the newly obtained ID.  Why is it necessary to come back to the controller just to make the call to update?